src/server/src/main.rs

```diff
@@ -1,23 +1,215 @@
+use bastion::prelude::*;
 use ntex::web;
+use ntex::web::{App, HttpRequest, HttpResponse, HttpServer, Responder};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use uuid::Uuid;
 
-#[web::get("/")]
-async fn index() -> impl web::Responder {
-    "Hello, World!"
+// user
+#[derive(Deserialize, Serialize, Clone)]
+struct User 
+{
+    username: String,
+    password: String,
+    email: Option<String>,
+    age: Option<u32>,
+    gender: Option<String>,
 }
+// sessionId -> child process reference
+type SessionMap = Arc<Mutex<HashMap<String, ChildRef>>>;
+// username -> User
+type UserMap = Arc<Mutex<HashMap<String, User>>>;
 
-#[web::get("/{name}")]
-async fn hello(name: web::types::Path<String>) -> impl web::Responder {
-    web::HttpResponse::Ok()
-        .content_type("text/plain")
-        //THIS IS FOR THE DEMO, DO NOT DO THIS IN PRODUCTION GET RID OF THIS SOON
-        .set_header(ntex::http::header::ACCESS_CONTROL_ALLOW_ORIGIN, "*")
-        .body(format!("Hello {}!", &name))
+fn spawn_user_session(username: String) -> ChildRef 
+{
+    Bastion::children(|children| 
+        {
+        children.with_exec(move |ctx| 
+            {
+            let username_clone = username.clone();
+            async move 
+            {
+                loop 
+                {
+                    let msg = ctx.recv().await?;
+                    if let Ok(task) = msg.downcast::<String>() 
+                    {
+                        println!("Session for {} doing: {}", username_clone, task);
+                    }
+                    ctx.reply(format!("done {}", username_clone)).unwrap();
+                }
+                Ok(())
+            }
+        })
+    })
+    .expect("Failed to spawn user session")
+}
+
+async fn create_user_handler(
+    body: web::types::Json<User>,
+    path: web::types::Path<(String,)>,
+    user_map: web::types::Data<UserMap>,) -> impl Responder 
+{
+    let new_user = body.into_inner();
```

Comment:
im curious, what are the semantics if the `web::types::Json` payload contains more stuff (or different stuff) than the `<User>` wants. i don't see any error handling done at this level (not even an `.unwrap()`), so where does it happen?